dask_cuda/tests/test_initialize.py

```diff
@@ -225,3 +227,195 @@ def test_dask_cuda_import():
+            cuda.core.experimental.Device().set_current()
+            print("Problematic library: Created CUDA context at import time")
+            os.environ['SUBPROCESS_CUDA_CONTEXT_CREATED'] = '1'
+        except Exception as e:
+            raise RuntimeError(
+                f"Problematic library: Failed to create CUDA context({e})"
+            )
+            os.environ['SUBPROCESS_CUDA_CONTEXT_CREATED'] = '0'
+        """
+    )
+
+    problematic_lib_path = os.path.join(temp_dir, "problematic_cuda_library.py")
+    with open(problematic_lib_path, "w") as f:
+        f.write(problematic_library_code)
+
+    # Create the main test script that imports the problematic library
+    # and creates LocalCUDACluster - this will run in a subprocess
+    main_script_code = textwrap.dedent(
+        f"""
+        # Main script that demonstrates the real-world problematic scenario
+        import os
+        import sys
+        import logging
+
+        # Add the temp directory to path so we can import our problematic library
+        sys.path.insert(0, '{temp_dir}')
+
+        print("=== Starting subprocess warnings test ===")
+
+        # This is the key part: import the problematic library BEFORE creating
+        # LocalCUDACluster. This creates a CUDA context that will be inherited
+        # by all worker subprocesses
+        print("Importing problematic library...")
+        import problematic_cuda_library
+
+        context_mode = os.environ.get('SUBPROCESS_CUDA_CONTEXT_CREATED', None)
+        if context_mode == "1":
+            print(f"Context creation successful")
+        else:
+            raise RuntimeError("Context creation failed")
+
+        if __name__ == "__main__":
+            try:
+                from dask_cuda import LocalCUDACluster
+                from dask_cuda.utils_test import IncreasedCloseTimeoutNanny
+
+                cluster = LocalCUDACluster(
+                    dashboard_address=None,
+                    worker_class=IncreasedCloseTimeoutNanny,
+                    protocol=f"{protocol}",
+                )
+                print("LocalCUDACluster created successfully!")
+
+                cluster.close()
+                print("Cluster closed successfully")
+
+            except Exception as e:
+                raise RuntimeError(f"Cluster setup error: {{e}}")
+
+        print("=== Subprocess warnings test completed ===")
+    """
+    )
+
+    main_script_path = os.path.join(temp_dir, "test_subprocess_warnings.py")
+    with open(main_script_path, "w") as f:
+        f.write(main_script_code)
+
+    try:
+        # Run the main script in a subprocess
+        result = subprocess.run(
+            ["/raid/pentschev/miniforge3/envs/rn-250904/bin/python", main_script_path],
+            capture_output=True,
+            text=True,
+            timeout=30,  # Reduced timeout for simpler test
+            cwd="/datasets/pentschev/src/dask-cuda",
```

Comment:
Maybe `os.getcwd()`? Though that might be already be the default, so maybe you need to adjust that somehow?

```suggestion
            cwd=os.getcwd(),
```